app.py

```diff
@@ -1,50 +1,140 @@
 from flask import Flask, render_template, request, redirect, url_for
+""" Flask-Login 
+Flask-Login provides user session management for Flask. It handles the common tasks of logging in, logging out, and remembering your users’ sessions over extended periods of time.
+https://flask-login.readthedocs.io/en/latest/
+"""
+from flask_login import LoginManager, login_required, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
+import requests
+import os
 import sys
 import logging
-from view_model import ViewModel
 
+from view_model import ViewModel
+from entity.user import User
+from entity.role import Role
+from entity.access_level import restricted
 from mongo_db.index import get_db_collection
 from mongo_db.db_queries import get_all_items, mark_item_as_complete, mark_item_as_uncomplete, mark_item_as_in_progress, add_new_item, remove_item
 
 date_time_format = "%Y-%m-%dT%H:%M:%S.%fZ"
 
 def create_app():
     app = Flask(__name__)
+    app.secret_key = os.environ.get("SECRET_KEY") or os.urandom(24)
+
+    login_disabled = os.getenv('LOGIN_DISABLED') == 'True'
+    app.config['LOGIN_DISABLED'] = login_disabled
 
     handler = logging.StreamHandler(sys.stdout)
     app.logger.addHandler(handler)
     app.logger.setLevel(logging.DEBUG)
 
     collection = get_db_collection()
 
+    # User session management setup
+    login_manager = LoginManager()
+
+    # This information is obtained upon registration of a new GitHub account
```

Comment:
It's a bit of a code smell if you feel these comments in the middle of your create_app function are helpful.
Some bits could be pulled out into other files/functions.